src/components/vm/vmMigrateDialog.jsx

```diff
@@ -0,0 +1,207 @@
+                       helperTextInvalid={validationFailed.destUri}
+                       value={destUri}
+                       placeholder={cockpit.format(_("Example, $0"), "qemu+ssh://192.0.2.16/system")}
+                       onChange={setDestUri} />
+        </FormGroup>
+    );
+};
+
+const OptionsRow = ({ temporary, live, setTemporary, setLive, setStorage }) => {
+    return (
+        <FormGroup label={_("Options")}
+                   aria-label={_("Options")}
+                   id="options"
+                   hasNoPaddingTop>
+            <Checkbox id="live"
+                      isChecked={live}
+                      label={_("Run after migration")}
+                      onChange={b => {
+                          setLive(b);
+                          if (!b)
+                              setStorage("shared");
+                      }} />
+            {live && <Checkbox id="temporary"
+                               isChecked={temporary}
+                               label={
+                                   <>
+                                       {_("Move temporarily")}
+                                       <Tooltip
+                                           position="top"
+                                           content={
+                                               <>
+                                                   <p>{_("When shut off on the destination, this virtual machine will return to the original host in a shut off state.")}</p>
+                                                   <p>{_("This feature is often used for hardware maintanance or load-balacing.")}</p>
+                                               </>}>
+                                           <OutlinedQuestionCircleIcon id="migrate-tooltip" />
+                                       </Tooltip>
+                                   </>
+                               }
+                               onChange={setTemporary} />
+            }
+        </FormGroup>
+    );
+};
+
+const StorageRow = ({ storage, setStorage, live }) => {
+    const copyDisabled = !live;
+    let copyRadio = (
+        <Radio id="copy"
+               name="source"
+               label={_("Copy storage")}
+               isDisabled={copyDisabled}
+               isChecked={storage === "copy"}
+               onChange={() => setStorage("copy")} />
+    );
+
+    if (copyDisabled) {
+        copyRadio = (
+            <Tooltip id="storage-copy-tooltip"
+                     content={_("Offline migration doesn't allow copying storage")}>
+                {copyRadio}
+            </Tooltip>
+        );
+    }
+
+    return (
+        <FormGroup label={_("Storage")}
+                   aria-label={_("Storage")}
+                   id="storage"
+                   hasNoPaddingTop>
+            <Radio id="shared"
+                   name="source"
+                   label={_("Use shared storage")}
```

Comment:
So this is a bit confusing here. I just read about VIR_MIGRATE_NON_SHARED_INC which is used when the VMs have a base image shared and only the top layer is copied. In this case we *do* have shared storage, but there is not a danger of data corruption, even if the VM is running on both host and destination.
 
In addition, I think this needs clarification, that shared storage means here that nothing will be copied over.